File: c#/Creational/Builder/Builder/EmployeeBuilder.cs
using Dto.Address;
using Dto.Phone;
using Dto.Contact;
using Dto.Employee;

namespace Builder.EmployeeBuilder
{
    public class EmployeeBuilderClass
    {
        private string? name = null;
        private int? age = null;
        private Gender? gender = null;
        private AddressDto? address = null;
        private List<PhoneDto> phones = new List<PhoneDto>();
        private List<ContactDto> contacts = new List<ContactDto>();


        public EmployeeBuilderClass SetName(string name)
        {
            this.name = name;
            return this;
        }

        public EmployeeBuilderClass SetAge(int age)
        {
            this.age = age;
            return this;
        }

        public EmployeeBuilderClass SetGender(Gender gender)
        {
            this.gender = gender;
            return this;
        }

        public EmployeeBuilderClass SetAddress(string addr, string city, string country, string cp)
        {
            address = new AddressDto(city, country, cp, addr);
            return this;
        }

        public EmployeeBuilderClass AddPhone(string phoneNumber, TypePhone phoneType)
        {
            phones.Add(new PhoneDto(phoneNumber, phoneType));
            return this;
        }
        
        public EmployeeBuilderClass AddContact(string name, string phoneNumber)
        {
            contacts.Add(new ContactDto(name, phoneNumber));
            return this;
        }

        public EmployeeDto Build()
        {
            if (string.IsNullOrEmpty(name) || !age.HasValue || !gender.HasValue || address == null)
            {
                throw new InvalidOperationException("Missing required fields to build Employee");
            }
            return new EmployeeDto(name, age.Value, gender.Value, address, phones, contacts);
        }
    }
}

File: typescript/src/creational/builder/builder/EmployeeBuilder.ts
import { Address } from "../dto/Address";
import { Phone, TYPE_PHONE } from "../dto/Phone";
import { Contact } from "../dto/Contact";
import { Employee, GENDER } from "../dto/Employee";

export class EmployeeBuilder {
    private name: string | null = null;
    private age: number | null = null;
    private gender: GENDER | null = null;
    private address: Address | null = null;
    private phones: Phone[] = [];
    private contacts: Contact[] = [];

    setName(name: string): this {
        this.name = name;
        return this;
    }

    setAge(age: number): this {
        this.age = age;
        return this;
    }

    setGender(gender: GENDER): this {
        this.gender = gender;
        return this;
    }

    setAddress(addr: string, city: string, country: string, cp: string): this {
        this.address = new Address(city, country, cp, addr);
        return this;
    }

    addPhone(phoneNumber: string, phoneType: TYPE_PHONE): this {
        this.phones.push(new Phone(phoneNumber, phoneType));
        return this;
    }

    addContact(name: string, phoneNumber: string): this {
        this.contacts.push(new Contact(name, phoneNumber));
        return this;
    }

    build(): Employee {
        if (!this.name || !this.age || !this.gender || !this.address) {
            throw new Error("Missing required fields to build Employee");
        }
        return new Employee(this.name, this.age, this.gender, this.address, this.phones, this.contacts);
    }
}